api/src/main/java/org/openmrs/module/shr/cdahandler/processor/util/LocationOrganizationProcessorUtil.java

```diff
@@ -121,11 +121,19 @@ private Location createLocation(Organization organization, String id) throws Doc
 	 */
 	public Location getOrganizationByExternalId(String id) throws DocumentImportException
 	{
+		try {
+			Integer locationId = Integer.valueOf(id.substring(0, id.indexOf("^")));
+			Location locationById = Context.getLocationService().getLocation(locationId);
+			if (locationById != null)
+				return locationById;
+		} catch (StringIndexOutOfBoundsException ex) {
```

Comment:
we should rather an if here, if (locationId == -1), then skip or something along those line

Reply:
Corrected